src/dmd/expression.h

```diff
@@ -95,7 +108,7 @@ class Expression : public ASTNode
     virtual bool checkType();
     virtual bool checkValue();
     bool checkDeprecated(Scope *sc, Dsymbol *s);
-    virtual int checkModifiable(Scope *sc, int flag = 0);
+    virtual int checkModifiable(Scope *sc, ModifyFlags flag =ModifyFlags.none);
```

Comment:
```suggestion
    virtual int checkModifiable(Scope *sc, ModifyFlags flag = ModifyFlags::none);
```

Reply:
These functions could really do with being devirtualized and made `extern(D)` in continuation of #12809 and others.

Reply:
Will look into it.